src/Arcus.Security.Core/CompositeSecretProvider.cs

```diff
@@ -160,32 +162,67 @@ public async Task InvalidateSecretAsync(string secretName)
                 throw new SecretNotFoundException(secretName, noRegisteredException);
             }
 
+            var results = new Collection<(string description, string key, string outcome)>();
```

Comment:
Yes, I added this bc when you add the secret name mutation to the secret store, your 'concrete' secret provider will get wrapped inside a `MutatedSecretNameSecretProvider` (much like with caching, and the `CachedSecretProvider`).

If we then throw an exception _"Can't find secret with name "bar" in provider "{Provider""_, we will have the `MutatedSecretNameSecretProvider` in the logging; making it (IMHO) somewhat useless bc you can't see the _original_ provider.

So, I created this interface `ISecretProviderDescription` that _any_ `ISecretProvider` can implement (ex. `KeyVaultSecretProvider` had `Azure Key Vault`). When I then use secret mutations and the Azure Key Vault secret provider, I can create this exception when the secret is not found:
`> Mutated + Azure Key Vault ('Arcus:Secret:Name' -> 'Arcus.Secret.Name') = NotFound`.

Reply:
I think it's a bit of overkill to be honest, can't we just use the original & mutated name somehow? I don't see how the current exception message is providing a lot of value, to be honest.

Reply:
Ok, I'll revert some, it was mostly bc you could see each secret store registration and could see exact which providers failed or couldn't find the secret + the mutation/cached composition.
But I'll revert some so you can only see the secret name mutation.

Reply:
Hmm, question, and I think that's the reason for this change: 
_why_ I made this was for writting the output of multiple providers.

So, let's say you have two providers with both mutations... and they both fail to retrieve the secret. Which one would you log? And if a provider without a mutation is added after those two and also fails, which secret name should then be added to the exception?
I think that's why I made this table of contents in the exception message so you can see all of it.

Reply:
Frankly, what we need here is logging inside the provider as a verbose trace for the metric that was not found.

The aggregate provider just throws secret not found with the original one and be done with it, no?

Reply:
Hmm, no bc we should check all sub-providers in the composite before we can say that the secret was not found. What if the first one throws not found and the second one finds the secret? Then you will want the secret. But if both the first one and the second one throws... That's why this setup. Open for alternatives of course.

Reply:
> Frankly, what we need here is logging inside the provider as a verbose trace for the metric that was not found.
> 
> The aggregate provider just throws secret not found with the original one and be done with it, no?

We already traced this, and throws the orignal secret name, it was just by your suggestion that we would maybe need the mutated secret name too that I changed this.

Reply:
Then I think it's ok because of the reasons you've mentioned above. We already provide insights on what was not found per provider (trace with mutated name) and throw an exception with original name.

If that is correct, then we can roll this change back - So sorry about this but didn't take this into account :(

Reply:
No problem, that's what PR comments are for, I'll update this after I've done with something else.

Reply:
Ok, that's done, and I added some examples with the other providers as well (user secrets and key vault).